### ISSUE TITLE
Is there any reason for using tensor.data?

### ISSUE THREAD

**Issue by minlee077** [pytorch/tutorials#606]
https://github.com/pytorch/tutorials/blob/60d6ef365e36f3ba82c2b61bf32cc40ac4e86c7b/beginner_source/blitz/autograd_tutorial.py#L160

This is an official tutorial codes.

Is there any reason for using tensor.data? 
~~~
x = torch.randn(3, requires_grad=True)

y = x * 2
while y.data.norm() < 1000:
    y = y * 2
~~~

If not, I think this code should be replaced by 
~~~
y = x * 2
while y.detach().norm() < 1000:
    y = y * 2
~~~
